plugins/capture.rb

```diff
@@ -0,0 +1,605 @@
+require 'uri'
+require 'rex/sync/event'
+require 'fileutils'
+
+module Msf
+
+class Plugin::HashCapture < Msf::Plugin
+
+  class ConsoleCommandDispatcher
+    include Msf::Ui::Console::CommandDispatcher
+
+    class CaptureJobListener
+      def initialize(name, done_event)
+        @name = name
+        @done_event = done_event
+      end
+
+      def waiting(id)
+        self.succeeded = true
+        print_good("#{@name} started")
+        @done_event.set
+      end
+
+      def start(id); end
+
+      def completed(id, result, mod); end
+
+      def failed(id, error, mod)
+        print_error("#{@name} failed to start")
+        @done_event.set
+      end
+
+      attr_accessor :succeeded
+
+    end
+
+    HELP_REGEX = /^-?-h(?:elp)?$/
+    @@stop_opt_parser = Rex::Parser::Arguments.new(
+      '--session' => [ true, 'Session to stop (otherwise all capture jobs on all sessions will be stopped)' ],
+      '-h' => [false, 'Display this message' ],
+    )
+
+    @@start_opt_parser = Rex::Parser::Arguments.new(
+      '--session' => [ true, 'Session to bind on' ],
+      '-i' => [ true, 'IP to bind to' ],
+      '--spoofip' => [ true, 'IP to use for spoofing (poisoning); default is the bound IP address' ],
+      '--regex' => [ true, 'Regex to match for spoofing' ],
+      '--basic' => [ false, 'Use Basic auth for HTTP listener (default is NTLM)' ],
+      '--cert' => [ true, 'Path to SSL cert for encrypted communication' ],
+      '--configfile' => [ true, 'Path to a config file' ],
+      '--logfile' => [ true, 'Path to store logs' ],
+      '--hashdir' => [ true, 'Directory to store hash results' ],
+      '--stdout' => [ false, 'Show results in stdout' ],
+      '-v' => [ false, 'Verbose output' ],
+      '-h' => [false, 'Display this message' ],
+    )
```

Comment:
We just recently added support for long form arguments in #16014.

I'd propose these changes to add some short and long form combos:

```suggestion
    @@stop_opt_parser = Rex::Parser::Arguments.new(
      '--session'      => [ true, 'Session to stop (otherwise all capture jobs on all sessions will be stopped)' ],
      ['-h', '--help'] => [ false, 'Display this message' ],
    )

    @@start_opt_parser = Rex::Parser::Arguments.new(
      '--session'         => [ true, 'Session to bind on' ],
      ['-i', '--ip']      => [ true, 'IP to bind to' ],
      '--spoofip'         => [ true, 'IP to use for spoofing (poisoning); default is the bound IP address' ],
      '--regex'           => [ true, 'Regex to match for spoofing' ],
      ['-b','--basic']    => [ false, 'Use Basic auth for HTTP listener (default is NTLM)' ],
      '--cert'            => [ true, 'Path to SSL cert for encrypted communication' ],
      '--configfile'      => [ true, 'Path to a config file' ],
      '--logfile'         => [ true, 'Path to store logs' ],
      '--hashdir'         => [ true, 'Directory to store hash results' ],
      '--stdout'          => [ false, 'Show results in stdout' ],
      ['-v', '--verbose'] => [ false, 'Verbose output' ],
      ['-h', '--help']    => [ false, 'Display this message' ],
    )
```

The `parse_start_args` method would need to be updated with the new flags. For `-b` would need to be added for `--basic` since the parser will prefer to pass the short-form.